src/pydata_sphinx_theme/assets/styles/sections/_announcement.scss

```diff
@@ -33,6 +33,31 @@
   a {
     color: var(--pst-color-inline-code-links);
   }
+
+  // The "Switch to stable version" link (styled like a button)
+  .button-link-to-stable-version {
+    @include box-shadow();
+
+    $background-color: var(--pst-color-danger);
+    $hover-background-color: var(--pst-color-danger-highlight);
+    $color: var(--pst-color-heading-color);
+
+    background-color: $background-color;
+    border-color: $background-color;
+    color: $color;
+    border-radius: 0.25rem;
+
+    &:hover {
+      background-color: $hover-background-color;
+      border-color: $hover-background-color;
+      color: $color;
+    }
+
+    &:focus-visible {
+      outline: $focus-ring-width solid $background-color;
+      outline-offset: $focus-ring-width;
```

Comment:
I don't think it's worth adding styles to take care of hover + focus-visible, especially since focus-visible is meant for keyboard-only use whereas hover is meant for mouse.